### PR TITLE
Allow KBaseAssembly read object types for backwards compatibility

### PR DESCRIPTION
**Pull request by dcchivian**
Addresses PUBLIC-254 and PTV-793

### CODE REVIEW

**Comment by dcchivian** [comment]
passing unit test

File: ui/narrative/methods/run_maxbin2/display.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #
 # define display information
 #
-name     : MaxBin2 Contig Binning - v2.2.3
+name     : MaxBin2 Contig Binning - v2.2.4
```
Comment (Tianhao-Gu) [inline ui/narrative/methods/run_maxbin2/display.yaml]: should be v1.1.0

File: lib/kb_maxbin/kb_maxbinImpl.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ class kb_maxbin:
     # state. A method could easily clobber the state set by another while
     # the latter method is running.
     ######################################### noqa
-    VERSION = "1.0.0"
+    VERSION = "1.0.1"
```
Comment (Tianhao-Gu) [inline lib/kb_maxbin/kb_maxbinImpl.py]: please run make. it should be updated to 1.1.0

File: lib/kb_maxbin/Utils/MaxBinUtil.py [hunk 3]
```diff
@@ -405,6 +405,7 @@ def run_maxbin(self, params):
         log('Saved result files to: {}'.format(result_directory))
         log('Generated files:\n{}'.format('\n'.join(os.listdir(result_directory))))
 
+        binned_contig_obj_ref = None
```
Comment (Tianhao-Gu) [inline lib/kb_maxbin/Utils/MaxBinUtil.py]: unnecessary initialization

File: ui/narrative/methods/run_maxbin2/spec.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-    "ver": "1.0.0",
+    "ver": "1.0.1",
```
Comment (Tianhao-Gu) [inline ui/narrative/methods/run_maxbin2/spec.json]: should be 1.1.0

File: ui/narrative/methods/run_maxbin2/spec.json [hunk 5]
```diff
@@ -42,7 +42,7 @@
             "default_values" : [ "" ],
             "field_type" : "text",
             "text_options" : {
-              "valid_ws_types" : ["KBaseFile.PairedEndLibrary", "KBaseFile.SingleEndLibrary"]
+              "valid_ws_types" : ["KBaseFile.PairedEndLibrary", "KBaseFile.SingleEndLibrary", "KBaseAssembly.PairedEndLibrary", "KBaseAssembly.SingleEndLibrary"]
```
Comment (Tianhao-Gu) [inline ui/narrative/methods/run_maxbin2/spec.json]: needs tests for KBaseAssembly.PairedEndLibrary and SingleEndLibrary objects

**Comment by Tianhao-Gu** [comment]
Besides comments, this PR also has conflicts with master.

File: lib/kb_maxbin/Utils/MaxBinUtil.py (hunk 3 above)
Comment (dcchivian) [reply lib/kb_maxbin/Utils/MaxBinUtil.py]: initialization to None is never bad and not a waste of code

File: lib/kb_maxbin/kb_maxbinImpl.py (hunk 2 above)
Comment (dcchivian) [reply lib/kb_maxbin/kb_maxbinImpl.py]: Actually, given that there's a version change of the wrapped tool, I agree that 1.1.0 is better than 1.0.1 as the updated module verison

File: ui/narrative/methods/run_maxbin2/display.yaml (hunk 1 above)
Comment (dcchivian) [reply ui/narrative/methods/run_maxbin2/display.yaml]: the App name when the wrapped tool itself has a version includes the version of the wrapped tool, not the module version

File: ui/narrative/methods/run_maxbin2/spec.json (hunk 5 above)
Comment (dcchivian) [reply ui/narrative/methods/run_maxbin2/spec.json]: Can we still upload KBaseAssembly types?

Comment (Tianhao-Gu) [reply ui/narrative/methods/run_maxbin2/spec.json]: I don't know. But if you want to include new types, you should add tests.

Comment (dcchivian) [reply ui/narrative/methods/run_maxbin2/spec.json]: I agree.  I can't use the read uploader though.  I'll have to go through the pain-in-the-neck process of creating KBaseAssembly type read libraries myself in the unit tests.  Hold on...

**Comment by dcchivian** [comment]
will sync from master and redo PR

**Pull request closed without merging**